### PR TITLE
Fixes csv issue

### PR DESCRIPTION
**Pull request by Nino-K**
- This fix is related to issue #32 where if the csv output is specified, Run
  gets blocked on the stopCh

### CODE REVIEW

**Comment by PatrikSteuer** [comment]
@rakyll can you review and merge this? Would be a great to get it running

File: requester/requester_test.go [hunk 1]
```diff
@@ -133,3 +135,45 @@ func TestBody(t *testing.T) {
...
+	req, _ := http.NewRequest("GET", server.URL, nil)
+	w := &Work{
+		Request: req,
+		N:       20,
+		C:       2,
+		Output:  "csv",
+	}
+	stdOut := readStdout(w.Run)
+	if count != 20 {
+		t.Errorf("Expected to work 20 times, found %v", count)
```
Comment (rakyll) [inline requester/requester_test.go]: Let's use got/want in the assertions. It makes it easier to refactor them in the future. Also follow got/want style error messaging.

if got, want := count, 20; got != want {
t.Errorf("count = %v; want %v", got, want)
}

File: requester/requester_test.go [hunk 2]
```diff
@@ -133,3 +135,45 @@ func TestBody(t *testing.T) {
...
+	}
+	scanner := bufio.NewScanner(bufio.NewReader(stdOut))
+	var csvData []string
+	for scanner.Scan() {
+		csvData = append(csvData, scanner.Text())
+	}
+	if csvData[0] != "response-time" {
+		t.Error("Expected csv response-time header")
```
Comment (rakyll) [inline requester/requester_test.go]: if got, want := csvData[0], "response-time"; got != want {
t.Errorf("cvs header[0] = %q; want %q", got, want)
}

File: requester/requester_test.go [hunk 3]
```diff
@@ -133,3 +135,45 @@ func TestBody(t *testing.T) {
...
+	}
+	if len(csvData) != 21 {
```
Comment (rakyll) [inline requester/requester_test.go]: if got, want := len(csvData), 21; got != want {
t.Errorf("number of lines of csv output = %v; want %v", got, want)
}

File: requester/requester.go [hunk 4]
```diff
@@ -135,8 +135,8 @@ func (b *Work) Run() {
 	}()
 
 	b.runWorkers()
-	stopCh <- struct{}{}
 	if b.Output == "" {
+		stopCh <- struct{}{}
```
Comment (rakyll) [inline requester/requester.go]: We also shouldn't initialize progress reporting if b.Output != "". I will try to rewrite the progress handling if you don't want to deal with the current mess.

**Comment by yqf3139** [comment]
Hi, I just ran into the same issue. It will be nice if the issue is solved in the master branch.

**Comment by rakyll** [comment]
Fixed via 2969c6a, please open an issue again if you experience something odd.

**Pull request closed without merging**